src/coreclr/scripts/antigen_setup.py

```diff
@@ -0,0 +1,158 @@
+
+import argparse
+from os import path
+import os
+from os import listdir
+from coreclr_arguments import *
+from superpmi_setup import run_command, copy_directory, set_pipeline_variable
+from superpmi import ChangeDir, TempDir
+import tempfile
+
+parser = argparse.ArgumentParser(description="description")
+
+parser.add_argument("-arch", help="Architecture")
+parser.add_argument("-platform", help="OS platform")
+parser.add_argument("-source_directory",  help="path to source directory")
+parser.add_argument("-core_root_directory",  help="path to CORE_ROOT directory")
+is_windows = platform.system() == "Windows"
+
+def setup_args(args):
+    """ Setup the args.
+
+    Args:
+        args (ArgParse): args parsed by arg parser
+
+    Returns:
+        args (CoreclrArguments)
+
+    """
+    coreclr_args = CoreclrArguments(args, require_built_core_root=False, require_built_product_dir=False,
+                                    require_built_test_dir=False, default_build_type="Checked")
+
+    coreclr_args.verify(args,
+                        "source_directory",
+                        lambda source_directory: os.path.isdir(source_directory),
+                        "source_directory doesn't exist")
+
+    coreclr_args.verify(args,
+                        "core_root_directory",
+                        lambda core_root_directory: os.path.isdir(core_root_directory),
+                        "core_root_directory doesn't exist")
+
+    coreclr_args.verify(args,
+                        "arch",
+                        lambda unused: True,
+                        "Unable to set arch")
+
+    coreclr_args.verify(args,
+                        "platform",
+                        lambda unused: True,
+                        "Unable to set platform")
+
+    return coreclr_args
+
+
+def main(main_args):
+    """Main entrypoint
+
+    Args:
+        main_args ([type]): Arguments to the script
+    """
+
+    coreclr_args = setup_args(main_args)
+    arch_name = coreclr_args.arch
+    os_name = "win" if coreclr_args.platform.lower() == "windows" else "linux"
+    run_configuration = "{}-{}".format(os_name, arch_name)
+    source_directory = coreclr_args.source_directory
+
+    # CorrelationPayload directories
+    correlation_payload_directory = path.join(coreclr_args.source_directory, "payload")
+    scripts_src_directory = path.join(source_directory, "src", "coreclr", 'scripts')
+    coreroot_dst_directory = path.join(correlation_payload_directory, "CoreRoot")
+    antigen_dst_directory = path.join(correlation_payload_directory, "exploratory")
+
+    helix_source_prefix = "official"
+    creator = ""
+    ci = True
+    if is_windows:
+        helix_queue = "Windows.10.Arm64" if arch_name == "arm64" else "Windows.10.Amd64.X86"
+    else:
+        if arch_name == "arm":
+            helix_queue = "(Ubuntu.1804.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7-bfcd90a-20200121150440"
```

Comment:
It seems problematic that we have Helix queue names hard-coded here, as well as in superpmi_setup.py, instead of using the centralized helix-queues-setup.yml file. If nothing else, the python scripts should share a small, simple, helix-queues-setup.py helper script if we can't/don't use the yml file. This change could be done as a follow-up.

Reply:
Sounds reasonable. Added in the list of https://github.com/dotnet/runtime/issues/59559